src/components/Navigator/NavigatorDataProvider.vue

```diff
@@ -83,6 +97,81 @@ export default {
         this.isFetching = false;
       }
     },
+    /**
+     * @param {{path: string, type: string, title: string, children?: [] }[]} childrenNodes
+     * @param {NavigatorFlatItem | null} parent
+     * @param {Number} depth
+     * @param {Boolean} parentBetaStatus
+     * @return {NavigatorFlatItem[]}
+     */
+    flattenNestedData(childrenNodes, parent = null, depth = 0, parentBetaStatus = false) {
```

Comment:
Lets move the `flattenNestedData` and `hashCode` into the utils. I think this will help thirdparty clients create their own data providers.

Reply:
Resolved [c17ff68](https://github.com/apple/swift-docc-render/pull/396/commits/c17ff680b73bf74237d8cf89054fcb7cd651b07a)